sycl/test-e2e/AddressSanitizer/common/kernel-debug.cpp

```diff
@@ -0,0 +1,17 @@
+// REQUIRES: linux, cpu
+// RUN: %{build} %device_sanitizer_flags -O2 -g -o %t
+// RUN: env SYCL_PREFER_UR=1 UR_LAYER_ASAN_OPTIONS=debug:1 %{run} %t 2>&1 | FileCheck --check-prefixes CHECK-DEBUG %s
+#include <sycl/sycl.hpp>
+
+int main() {
+  sycl::queue Q;
+  int *array = sycl::malloc_device<int>(1, Q);
+
+  Q.submit([&](sycl::handler &h) {
+    h.single_task<class MyKernel>([=]() { *array = 0; });
+  });
+  Q.wait();
+  // CHECK-DEBUG: [kernel]
```

Comment:
What is the purpose of this test?

Reply:
This test is used to test `UR_LAYER_ASAN_OPTIONS=debug:1` runtime flag.

Reply:
And what is it supposed to be, exactly? Imagine someone not familiar with sanitizer implementation reading this test, what could they find?

Are you even sure this single line should be tested at all?

Reply:
Also, `array` leaks.

Reply:
> And what is it supposed to be, exactly? Imagine someone not familiar with sanitizer implementation reading this test, what could they find?
> 
> Are you even sure this single line should be tested at all?

Makes sense. 
Currently, `UR_LAYER_ASAN_OPTIONS=debug:1` is more useful for us (sanitizer developers). 
Besides, we're going to remove some of messages in release build for performance.

Okay, I'll remove this test!